### ISSUE TITLE
ExplainerDashboard fails with pandas==3.4.0

### ISSUE THREAD

**Issue by tvdboom** [oegedijk/explainerdashboard#175]
Hi,

I noticed that the dashboard fails for classifications since pandas version 3.4.0. Not sure if it's something between pandas and plotly or related to this package.

Minimum example:

```
from sklearn.datasets import load_breast_cancer
from sklearn.ensemble import RandomForestClassifier
from explainerdashboard import ClassifierExplainer, ExplainerDashboard

X, y = load_breast_cancer(return_X_y=True, as_frame=True)
explainer = ClassifierExplainer(RandomForestClassifier().fit(X, y), X, y)
dashboard = ExplainerDashboard(explainer=explainer)
```

The resulting error is:

```
Traceback (most recent call last):
  File "C:/Users/Mavs/Documents/Python/ATOM/run_automl.py", line 136, in <module>
    dashboard = ExplainerDashboard(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\dashboards.py", line 575, in __init__
    self.explainer_layout = ExplainerTabsLayout(explainer, tabs, title,
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\dashboards.py", line 95, in __init__
    self.downloadable_tabs = [tab for tab in self.tabs if tab.to_html(add_header=False) != "<div></div>"]
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\dashboards.py", line 95, in <listcomp>
    self.downloadable_tabs = [tab for tab in self.tabs if tab.to_html(add_header=False) != "<div></div>"]
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\dashboard_components\composites.py", line 194, in to_html
    [to_html.hide(self.precision.to_html(state_dict, add_header=False), hide=self.hide_precision),
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\dashboard_components\classifier_components.py", line 639, in to_html
    fig = self.explainer.plot_precision(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\explainers.py", line 57, in inner
    return func(self, *args, **kwargs)
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\explainers.py", line 3009, in plot_precision
    return plotly_precision_plot(precision_df,
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\explainerdashboard\explainer_plots.py", line 257, in plotly_precision_plot
    trace1 = go.Bar(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\plotly\graph_objs\_bar.py", line 3166, in __init__
    self["width"] = _v
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\plotly\basedatatypes.py", line 4819, in __setitem__
    self._set_prop(prop, value)
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\plotly\basedatatypes.py", line 5163, in _set_prop
    raise err
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\plotly\basedatatypes.py", line 5158, in _set_prop
    val = validator.validate_coerce(val)
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\_plotly_utils\basevalidators.py", line 770, in validate_coerce
    self.raise_invalid_val(v)
  File "C:\Users\Mavs\Documents\Python\ATOM\venv\lib\site-packages\_plotly_utils\basevalidators.py", line 289, in raise_invalid_val
    raise ValueError(
ValueError: 
    Invalid value of type 'pandas.core.series.Series' received for the 'width' property of bar
        Received value: 0    0.09
0    0.09
0    0.09
0    0.09
0    0.09
0    0.09
0    0.09
0    0.09
0    0.09
0    0.09
Name: bin_width, dtype: object
    The 'width' property is a number and may be specified as:
      - An int or float in the interval [0, inf]
      - A tuple, list, or one-dimensional numpy array of the above
```

**Comment by srj-surajchauhan** [comment]
facing the same issue, is there any solution for the this?

**Comment by srj-surajchauhan** [comment]
facing the same issue, is there any solution for the this?
my pandas version is 1.4.0 and some of the dash feature are also showing deprecated

temp-fix:Problem is with new pandas version it is working fine on pandas version 1.2.1

**Comment by oegedijk** [comment]
Hi, @tvdboom and @srj-surajchauhan, apologies I have had zero time lately to look into this. Should have some time later this week though. Nightly tests are also failing due to this reason so agree that need to find a fix ASAP.

Cheers

**Comment by oegedijk** [comment]
okay, found the bug. The latest version of Pandas turned that column `dtype` into `obj` for some reason, while it should be floats. Will do some additional fixes for dash 2.0 compatibility and then do a quick release.

**Comment by tvdboom** [comment]
The newest version (0.3.8) fixes this issue. Closing it.
